proof/file.go

```diff
@@ -203,8 +231,8 @@ func (f *File) NumProofs() int {
 // ProofAt returns the proof at the given index. If the file is empty, this
 // returns nil.
 func (f *File) ProofAt(index uint32) (*Proof, error) {
-	if f.IsEmpty() {
-		return nil, ErrNoProofAvailable
+	if err := f.IsValid(); err != nil {
```

Comment:
I think we also want a version in the proof itself? https://github.com/lightninglabs/taproot-assets/blob/main/proof/proof.go#L43

So one top level is a version that describes how to parse and the serialization, etc. The other is a version for each proof that maybe tells how to validate, or which fields are expected to be there, etc.

Reply:
Added that as a last commit.